networkx/algorithms/centrality/second_order_centrality.py

```diff
@@ -106,6 +109,20 @@ def second_order_centrality(G):
         raise nx.NetworkXException("Empty graph.")
     if not nx.is_connected(G):
         raise nx.NetworkXException("Non connected graph.")
+    if [True for (i, j, d) in G.edges(data=True)
+            if 'weight' in d and d['weight'] < 0]:
+        raise nx.NetworkXException("Graph has negative edge weights.")
+
+    def unbias_graph(G):
```

Comment:
Why create a function for ```unbias_graph``` when you only use it once? You can just include the code directly.